File: packages/core/android/src/main/java/expo/modules/gaodemap/overlays/HeatMapViewModule.kt
package expo.modules.gaodemap.overlays

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * HeatMap 视图 Module
 */
class HeatMapViewModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("HeatMapView")

    View(HeatMapView::class) {
      Prop<List<Any>?>("data") { view: HeatMapView, data ->
        view.setData(data)
      }

      Prop<Boolean>("visible") { view: HeatMapView, visible ->
        view.setVisible(visible)
      }
      
      Prop<Int>("radius") { view: HeatMapView, radius ->
        view.setRadius(radius)
      }
      
      Prop<Double>("opacity") { view: HeatMapView, opacity ->
        view.setOpacity(opacity)
      }

      Prop<Map<String, Any>?>("gradient") { _: HeatMapView, _ ->
          // iOS only, ignore on Android
      }

      Prop<Boolean>("allowRetinaAdapting") { _: HeatMapView, _ ->
        // iOS only, ignore on Android
      }
    }
  }
}

File: packages/core/ios/overlays/HeatMapViewModule.swift
import ExpoModulesCore

public class HeatMapViewModule: Module {
    public func definition() -> ModuleDefinition {
        Name("HeatMapView")
        
        View(HeatMapView.self) {
            Prop("data") { (view: HeatMapView, data: [[String: Any]]) in
                view.setData(data)
            }

            Prop("visible") { (view: HeatMapView, visible: Bool) in
                view.setVisible(visible)
            }
            
            Prop("radius") { (view: HeatMapView, radius: Int) in
                view.setRadius(radius)
            }
            
            Prop("opacity") { (view: HeatMapView, opacity: Double) in
                view.setOpacity(opacity)
            }
            
            Prop("gradient") { (view: HeatMapView, gradient: [String: Any]?) in
                view.setGradient(gradient)
            }
            
            Prop("allowRetinaAdapting") { (view: HeatMapView, allow: Bool) in
                view.setAllowRetinaAdapting(allow)
            }
        }
    }
}
